src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -845,8 +887,29 @@ Microsoft::WRL::ComPtr<IDWriteTextFormat> DxFontRenderData::_BuildTextFormat(con
     ::Microsoft::WRL::ComPtr<IDWriteTextFormat3> format3;
     if (!_axesVector.empty() && !FAILED(format->QueryInterface(IID_PPV_ARGS(&format3))))
     {
+        std::optional<float> oldWeight;
+        if (!_useUserWeight)
+        {
+            // Remove the user set weight if we were told not to use it
+            // This is quite awkward, we don't want to permanently delete the weight axis from our _axesVector,
+            // we only want to delete it from the list we are about to pass into the text format
+            for (auto iter = _axesVector.begin(); iter != _axesVector.end(); ++iter)
+            {
+                if (iter->axisTag == DWRITE_FONT_AXIS_TAG_WEIGHT)
+                {
+                    oldWeight = iter->value;
+                    _axesVector.erase(iter);
+                    break;
+                }
+            }
+        }
         DWRITE_FONT_AXIS_VALUE const* axesList = _axesVector.data();
         format3->SetFontAxisValues(axesList, gsl::narrow<uint32_t>(_axesVector.size()));
+        // If we removed the user weight, make sure to put it back
+        if (oldWeight)
+        {
+            _axesVector.emplace_back(DWRITE_FONT_AXIS_VALUE{ DWRITE_FONT_AXIS_TAG_WEIGHT, oldWeight.value() });
+        }

```

Comment:
I'm generally quite against any code that takes the format, "Delete something from global state, then re-add it after the patrols have passed by". There is a better way to do this, and I suspect that we should do that instead.

Reply:
Changed!